src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -576,14 +569,33 @@ FancyZones::OnKeyDown(PKBDLLHOOKSTRUCT info) noexcept
             }
         }
     }
-    // Temporarily disable Win+Ctrl+Number functionality
-    //else if (m_inMoveSize && (info->vkCode >= '0') && (info->vkCode <= '9'))
-    //{
-    //    // This allows you to cycle through ZoneSets while dragging a window
-    //    Trace::FancyZones::OnKeyDown(info->vkCode, win, false /*control*/, true /*inMoveSize*/);
-    //    CycleActiveZoneSet(info->vkCode);
-    //    return false;
-    //}
+
+    if (m_settings->GetSettings()->quickLayoutSwitch)
+    {
+        int digitPressed = -1;
+        if ('0' <= info->vkCode && info->vkCode <= '9')
+        {
+            digitPressed = info->vkCode - '0';
+        }
+        else if (VK_NUMPAD0 <= info->vkCode && info->vkCode <= VK_NUMPAD9)
+        {
+            digitPressed = info->vkCode - VK_NUMPAD0;
+        }
+
+        bool dragging = m_windowMoveHandler.IsDragEnabled();
+        bool changeLayoutWhileNotDragging = !dragging && !shift && win && ctrl && alt && digitPressed != -1;
+        bool changeLayoutWhileDragging = dragging && digitPressed != -1;

```

Comment:
We should enabled the single key switch while dragging a window, even if the zones are not activated, the reason is that if the user has shift-activation enabled, its hard to press shift+digit for digits from 5 to 0.
In that cause we should flash the zones since they are not visible.